smaug/python/ops/control_flow_ops.py

```diff
@@ -30,9 +28,100 @@ def merge(input_tensors, name="merge"):
   Returns:
     A tensor that the available input tensor forwards to.
   """
-  return add_node(
-      name=name,
-      op=Merge,
-      input_tensors=input_tensors,
+  return common.add_node(
+      name=name, op=types_pb2.Merge, input_tensors=input_tensors,
       output_tensors_dims=[input_tensors[0].shape.dims],
       output_tensor_layout=input_tensors[0].shape.layout)[0]
+
+class ControlFlowContext:
+  def __init__(self, pred, branch):
+    """Construct a control flow context.
+
+    Args:
+      pred: Predication tensor.
+      branch: Value of 0 or 1 to denote the False/True branch of the control
+        flow.
+    """
+    self._pred = pred
+    self._branch = branch
+    # Names of all nodes that are within this context.
+    self._internal_nodes = set()
+    # Mappings of the name of a known external tensors to its switch node ouput.
```

Comment:
Had trouble parsing this comment, can you clarify?

Reply:
This keeps track of the external tensors to this context. The dict maps an external tensor's name to its corresponding switch node's output tensor.